### PR TITLE
Research page updates

### PR DESCRIPTION
**Pull request by vincerubinetti**
- change research page python build script to use citation data that is already in output, if it exists, to avoid running manubot and save time
- add thumbnails to rest of papers
- remove paper filter checkboxes/buttons in favor of just using search box
- show X of N papers info when searching
- populate search field from url on load
- debounce search box for better performance,
- have cleaner fallback for mark.js
- create thumbnail placeholder and make all paper cards have thumbnails
- make authors and publisher/data rows in paper cards be focusable, and expand when focused
- add personalized link to research page from member bio pages
- fix image links in one of the blog posts
- css tweaks
- change research output from json to yaml to save a bit of space

### CODE REVIEW

**Review by cgreene — APPROVED** [review]
This looks really nice!

**Comment by vincerubinetti** [comment]
@dongbohu looking for review on python script

File: _data/build-research.py [hunk 1]
```diff
@@ -16,46 +16,64 @@
...
+except Exception:
+    print("No input file provided.")
 
-# run manubot for each paper to get citation metadata
-citations = []
+# load existing output papers as yaml
+try:
+    with open(output_file, encoding="utf8") as file:
+        output_papers = yaml.load(file, Loader=yaml.FullLoader)
+except Exception:
```
Comment (dongbohu) [inline _data/build-research.py:29]: Same issue with the exception type.

**Review by dongbohu — COMMENTED** [review]
I left a few comments on the Python script.

File: _data/build-research.py [hunk 2]
```diff
@@ -16,46 +16,64 @@
...
-    # paper url
-    new_paper["url"] = citation.get("URL", "")
-    # add to list
-    new_papers.append(new_paper)
+    print("\n------------------------------")
+    print("Paper " + str(index) + " of " + str(len(input_papers)))
+    print("------------------------------\n")
+
+    # if input paper already exists in output, use that citation info to save time
+    existing = next((output_paper for output_paper in output_papers if input_paper.get("id", "") == output_paper.get("id", "")), None)
```
[the remark this answers is not in the document]
Comment (vincerubinetti) [reply _data/build-research.py:41]: What is a pythonic way to split this into multiple lines. Should I just do a for loop.

File: _data/build-research.py [hunk 3]
```diff
@@ -16,46 +16,64 @@
 output_file = os.path.join(current_dir, "research-output.json")
 
 # load input papers as yaml
-with open(input_file, encoding="utf8") as file:
-    input_papers = yaml.load(file, Loader=yaml.FullLoader)
+try:
+    with open(input_file, encoding="utf8") as file:
+        input_papers = yaml.load(file, Loader=yaml.FullLoader)
+except Exception:
+    print("No input file provided.")
```
[the remark this answers is not in the document]
Comment (vincerubinetti) [reply _data/build-research.py:23]: Perhaps I should just keep it as `Exception` and change the error message to something like "Problem with the input file".

File: _data/build-research.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (dongbohu) [reply _data/build-research.py:41]: Probably something like this:
```python
matches = [p for p in output_papers if p.get("id", "") == input_paper.get("id", "")]
if len(matches) > 0:
    ...
```

File: _data/build-research.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (dongbohu) [reply _data/build-research.py:23]: Yes, that would make more sense.

**Pull request merged**